### PR TITLE
Add support for token login (for bots)

### PR DESCRIPTION
**Pull request by maxpowa**
Allows you to log in with bot accounts, using 'token' as the email and the bot's token as the password.

Ex (obv invalid token):
```
client.run('token', 'SDMSDNZIQ123Dsna1243sdfk.DcGdhR.A3245ASFIOAI12ASDF_121')
```

### CODE REVIEW

**Comment by b1naryth1ef** [comment]
TBH this should be a separate function altogether, and perhaps @Rapptz can throw the old user/pass auth in a separate lib :^)

**Comment by AraHaan** [comment]
I did makke a commit for this however Danny turned it down sadly. https://github.com/Rapptz/discord.py/pull/134/files#diff-5bab608907a27a2180166115e6299cef
Be careful though if you change it in your latest copy of client.py to back it up Also all of your bots who are not on the new account type must have ``None`` on the token arg option for them to work.
But If it is the new account type the ``email`` and ``password`` args can be ``None`` and be ok.

**Comment by maxpowa** [comment]
This change doesn't break any existing behavior. It simply adds support for token auth in a non-invasive manner. This does not do any of the conversion from normal user account to bot account, so it's on the user to convert an account on their own (and as such 100% their problem if they convert their primary account to a bot account).

Looking at your PR, it would have broken existing implementations.

**Comment by AraHaan** [comment]
I edited my code since the PR to add in token checking (from what I hear from some ppl it is always a length of 27 characters so, I did this.

https://github.com/AraHaan/discord.py/blob/master/discord/client.py#L580-#L644

**Comment by uniquoooo** [comment]
@AraHaan You can send a request to `GET /api/auth/login` with the `authorization` header set and if it returns 400 Bad Request the token is invalid. If it succeeds it returns the token.

**Comment by maxpowa** [comment]
@AraHaan please keep the comments relative to this PR and not others to reduce confusion.

I'm not really sure if it's necessary but it wouldn't be very difficult to change the validitiy test in this PR to use `GET /api/auth/login` instead of `GET /api/users/@me`.

File: discord/client.py [hunk 1]
```diff
@@ -464,6 +464,24 @@ def login(self, email, password):
...
 
+        if email == "token":
+            log.info('logging in using static token')
+            self.token = password
+            self.headers['authorization'] = 'Bot {}'.format(self.token)
+            resp = yield from self.session.get(endpoints.ME, headers=self.headers)
+            log.debug(request_logging_format.format(method='GET', response=resp))
+
+            if resp.status != 200:
+                yield from resp.release()
```
Comment (khazhyk) [inline discord/client.py]: You never release the response if it is successful - my debug log has been complaining about this. :)

**Comment by Rapptz** [comment]
I did this a bit separately but I merged your work anyway. Just do `client.login('token')`.

**Pull request closed without merging**